src/librustdoc/clean/mod.rs

```diff
@@ -2491,7 +2505,23 @@ impl Clean<BareFunctionDecl> for hir::BareFnTy {
                 type_params: Vec::new(),
                 where_predicates: Vec::new()
             },
-            decl: (&*self.decl, &[][..]).clean(cx),
```

Comment:
The existing impls are *already* powerful enough for this (unless I wasn't careful), all you need is to replace `&[][..]` with `&self.arg_names`.

Reply:
Not exactly:

```
no method named `clean` found for type `(&rustc::hir::FnDecl, &syntax::ptr::P<[syntax::codemap::Spanned<syntax::ast::Symbol>]>)` in the current scope
    --> src/librustdoc/clean/mod.rs:2507:50
     |
2507 |             decl: (&*self.decl, &self.arg_names).clean(cx),
     |                                                  ^^^^^
     |
     = note: the method `clean` exists but the following trait bounds were not satisfied:
             `(&[syntax::ptr::P<rustc::hir::Ty>], &syntax::ptr::P<[syntax::codemap::Spanned<syntax::ast::Symbol>]>) : clean::Clean<clean::Arguments>`
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following trait defines an item `clean`, perhaps you need to implement it:
             candidate #1: `clean::Clean`
```

Reply:
Sounds like you need `&self.arg_names[..]` instead?

Reply:
Like, look for `ForeignItemFn`, it starts off with the same information but it *doesn't* implement the cleaning itself.